### PR TITLE
setup.py: Fail out if there are no prebuilts

### PR DESCRIPTION
**Pull request by Grazfather**

### CODE REVIEW

File: src/setup.py [hunk 1]
```diff
@@ -27,6 +27,9 @@
 pkg_info = os.path.join(package_dir, "PKG-INFO")
 in_source_package = os.path.isfile(pkg_info)
 if in_source_package:
+    if (sys.version_info[0] == 3 and sys.version_info[1] < 5) or \
+       (sys.version_info[0] == 2 and sys.version_info[1] < 7):
+        raise Exception("Frida only provides prebuilts for python 2.7 and 3.5")
```
Comment (oleavr) [inline src/setup.py]: We do provide a prebuild for 2.6 on Mac. Maybe it would be better to just try the urlopen and handle the specific failure?

Comment (Grazfather) [inline src/setup.py]: That's now I was doing it (and re-raising) but the stack trace sucked.

```
Traceback (most recent call last):
  File "setup.py", line 33, in <module>
    raise Exception("Frida only provides prebuilts for python 2.7 and 3.5")
Exception: Frida only provides prebuilts for python 2.7 and 3.5
```

vs

```running install
running bdist_egg
running egg_info
writing dependency_links to frida.egg-info/dependency_links.txt
writing entry points to frida.egg-info/entry_points.txt
writing top-level names to frida.egg-info/top_level.txt
writing requirements to frida.egg-info/requires.txt
writing frida.egg-info/PKG-INFO
reading manifest file 'frida.egg-info/SOURCES.txt'
writing manifest file 'frida.egg-info/SOURCES.txt'
installing library code to build/bdist.macosx-10.6-x86_64/egg
running install_lib
running build_py
warning: build_py: byte-compiling is disabled, skipping.

running build_ext
downloading prebuilt extension from https://pypi.python.org/packages/3.4/f/frida/frida-7.0.8-py3.4-macosx-10.6-intel.egg
trying to get prebuilt extension from the current folder file:///frida-7.0.8-py3.4-macosx-10.6-intel.egg
<urlopen error [Errno 2] No such file or directory: '/frida-7.0.8-py3.4-macosx-10.6-intel.egg'>
a'astasrt
Traceback (most recent call last):
  File "setup.py", line 74, in build_extension
    egg_data = urlopen(egg_url).read()
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/urllib/request.py", line 153, in urlopen
    return opener.open(url, data, timeout)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/urllib/request.py", line 461, in open
    response = meth(req, response)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/urllib/request.py", line 571, in http_response
    'http', request, response, code, msg, hdrs)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/urllib/request.py", line 499, in error
    return self._call_chain(*args)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/urllib/request.py", line 433, in _call_chain
    result = func(*args)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/urllib/request.py", line 579, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 404: Not Found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/urllib/request.py", line 1343, in open_local_file
    stats = os.stat(localfile)
FileNotFoundError: [Errno 2] No such file or directory: '/frida-7.0.8-py3.4-macosx-10.6-intel.egg'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "setup.py", line 78, in build_extension
    egg_data = urlopen(egg_url).read()
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/urllib/request.py", line 153, in urlopen
    return opener.open(url, data, timeout)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/urllib/request.py", line 455, in open
    response = self._open(req, data)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/urllib/request.py", line 473, in _open
    '_open', req)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/urllib/request.py", line 433, in _call_chain
    result = func(*args)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/urllib/request.py", line 1321, in file_open
    return self.open_local_file(req)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/urllib/request.py", line 1361, in open_local_file
    raise URLError(exp)
urllib.error.URLError: <urlopen error [Errno 2] No such file or directory: '/frida-7.0.8-py3.4-macosx-10.6-intel.egg'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "setup.py", line 141, in <module>
    'build_ext': FridaPrebuiltExt
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/Users/graziano/code/tempvenv/lib/python3.4/site-packages/setuptools/command/install.py", line 74, in run
    self.do_egg_install()
  File "/Users/graziano/code/tempvenv/lib/python3.4/site-packages/setuptools/command/install.py", line 89, in do_egg_install
    self.run_command('bdist_egg')
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/Users/graziano/code/tempvenv/lib/python3.4/site-packages/setuptools/command/bdist_egg.py", line 185, in run
    cmd = self.call_command('install_lib', warn_dir=0)
  File "/Users/graziano/code/tempvenv/lib/python3.4/site-packages/setuptools/command/bdist_egg.py", line 171, in call_command
    self.run_command(cmdname)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/Users/graziano/code/tempvenv/lib/python3.4/site-packages/setuptools/command/install_lib.py", line 20, in run
    self.build()
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/distutils/command/install_lib.py", line 107, in build
    self.run_command('build_ext')
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/Users/graziano/code/tempvenv/lib/python3.4/site-packages/setuptools/command/build_ext.py", line 52, in run
    _build_ext.run(self)
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/distutils/command/build_ext.py", line 348, in run
    self.build_extensions()
  File "/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/distutils/command/build_ext.py", line 457, in build_extensions
    self.build_extension(ext)
  File "setup.py", line 82, in build_extension
    raise Exception("Could not find prebuilt extension. Prebuilts only provided for 2.6-2.7 and 3.5")
Exception: Could not find prebuilt extension. Prebuilts only provided for 2.6-2.7 and 3.5
```

Comment (oleavr) [inline src/setup.py]: Ahh oops. I guess we can set a flag in the `catch`, and check for this flag afterwards, and throw there instead?

Comment (Grazfather) [inline src/setup.py]: This is a python 3 thing (implied chained exceptions). I can suppress it with `raise Exception(...) from None`, but then I need to check for the python version.

**Comment by Grazfather** [comment]
#71 does this better.

**Pull request closed without merging**